boards/sim/sim/sim/scripts/Make.defs

```diff
@@ -149,8 +149,13 @@ ifeq ($(CONFIG_SIM_M32),y)
   ARCHCFLAGS += -m32
   ARCHCXXFLAGS += -m32
 else
-  ARCHCFLAGS += -fno-pic -mcmodel=medium
-  ARCHCXXFLAGS += -fno-pic -mcmodel=medium
+  ifeq ($(CONFIG_HOST_MACOS),y)
+    ARCHCFLAGS += -fno-pic
```

Comment:
why not move -fno-pic out of ifeq/else to simplify the code?

Reply:
because i happened to feel this one simpler.
anyway i can change it for your preference.